lib/astutils.cpp

```diff
@@ -369,8 +369,8 @@ bool isOppositeCond(bool isNot, bool cpp, const Token * const cond1, const Token
     }
 
     // is condition opposite?
-    return ((comp1 == "==" && comp2 == "!=") ||
-            (comp1 == "!=" && comp2 == "==") ||
+    return ((comp1 == "==" && (comp2 == "!=" || comp2 == ">" || comp2 == "<")) ||
```

Comment:
this is ok when `isNot` is false. If that is true then this function should return true if comp1 is !comp2.